.github/workflows/sync-core-deps.yaml

```diff
@@ -0,0 +1,60 @@
+name: Sync k6 core dependencies
+
+on:
+  schedule:
+    - cron: 3 14 * * *  # Every day at 3:14
+  workflow_dispatch: {}
+
+jobs:
+  sync-k6-deps:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    permissions:
+      contents: write  # Push branch.
+      pull-requests: write  # Create PR.
+    env:
+      GH_TOKEN: ${{ github.token }}
+      BRANCH_NAME: ci-sync-deps
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+      - name: Sync dependencies
+        run: |
+          git checkout -b "$BRANCH_NAME"
+          go run ./hack/depsync 2> deps.log > depsync.sh
+
+          bash depsync.sh 2>&1 | tee -a go-get.log
+          go mod tidy
+
+      - name: Create PR
+        run: |
+          if [[ -z "$(git status --porcelain --untracked-files=no)" ]]; then
+            echo "Nothing to do"
+            exit 0
+          fi
+
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git commit -am "go.mod: sync dependencies with k6 core"
+          git push -f -u origin "$BRANCH_NAME"
+          
+          cat <<EOF > pr-body.txt
```

Comment:
I'm not a fan of embedding texts in this way.  I would prefer to have a script that generates this output.

Reply:
I'm not sure I follow. WDYM by a script that generates the output? Something like this same code but in a separate `.sh` file?

Reply:
Yes, exactly.

Reply:
What would be the advantage in this case? I agree the heredoc is not particularly pretty, but to me moving it to a separate script will make the yaml file harder to follow, as you would need to go somewhere else to check what the file is doing. Moreover such script wouldn't make sense outside of the context of this particular CI step.

Reply:
Basically, I think it is a bad practice to put any logic that does not depend on the CI environment (e.g. retrieving information provided by the CI) in the CI. Using a separate script allows running and testing this logic locally, if needed.

Reply:
Makes sense. I've added a `depsync.sh` script that does pretty much everything that is not related to git. LMK if this looks better :)